tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $registration_content = $this->assertSession()->elementExists('css', '#event-registration-block');
+    $this->assertRegisterButtonEnabled($registration_content);
+
+    // Assert "Registration date" field in the past.
+    $registration_start_date = (clone $static_time)->modify('- 2 months');
+    $registration_end_date = (clone $static_time)->modify('- 1 months');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $this->assertSession()->elementNotExists('css', 'a.ecl-u-mt-2xl.ecl-link.ecl-link--cta', $registration_content);
     $registration_info_content = $this->assertSession()->elementExists('css', 'p.ecl-u-type-paragraph.ecl-u-type-color-grey-75');
-    $this->assertEquals('Registration will open today, 2 February 2020, 08:00.', $registration_info_content->getText());
+    $this->assertEquals('Registration period ended on Saturday 18 January 2020, 01:00', $registration_info_content->getText());
 
-    $this->setTimezone('America/New_York');
-    \Drupal::entityTypeManager()->getViewBuilder('node')->resetCache([$node]);
+    // Assert "Registration date" field when registration is in progress.
+    $registration_start_date = (clone $static_time)->modify('- 10 days');
+    $registration_end_date = (clone $static_time)->modify('+ 10 days');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
     $this->drupalGet($node->toUrl());
-    $this->assertEquals('Registration will open today, 1 February 2020, 16:00.', $registration_info_content->getText());
 
-    // Assert registration date description when registration will end today.
-    $registration_start_date = (clone $static_time)->modify('-10 days');
-    $registration_end_date = (clone $static_time)->modify('+1 hours');
+    $this->assertRegisterButtonEnabled($registration_content);
+    $this->assertEquals('Book your seat, 1 week left to register, registration will end on 28 February 2020, 01:00', $registration_info_content->getText());
+
+    // Assert "Registration date" field when registration will finish today.
+    $registration_start_date = (clone $static_time)->modify('- 10 days');
+    $registration_end_date = (clone $static_time)->modify('+ 1 hours');
     $node->set('oe_event_registration_dates', [
       'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
       'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
     ])->save();
```

Comment:
See above, when testing time-related features we must never change the actual event dates, we should change time, run cron and reload the page, instead.